Add contributing guide (#388)
solid/specification: CONTRIBUTING.md

# Solid Technical Reports Contributing Guide

Thank you for investing your time into contributing to the Solid project!

This repository contains the source code of the work on [Solid Technical
Reports](https://solidproject.org/TR/) (TR) of the [W3C Solid Community
Group](https://www.w3.org/community/solid/) (CG) to meet the needs of the [Solid
Project](https://solidproject.org/).

The TRs include specifications, use cases and requirements, best practices and
guidelines, primers and notes about the Solid ecosystem.

## Code of conduct

Read the [Solid Code of
Conduct](https://github.com/solid/process/blob/main/code-of-conduct.md) and the
[Positive Work Environment at W3C: Code of Ethics and Professional
Conduct](https://www.w3.org/Consortium/cepc/) to keep our community approachable
and respectable.

## Contributions

> All substantive contributors must be members of the CG. It’s easy to [join the

Reviewer: Maybe reverse that first sentence? "In order to be a substantive contributor of the spec, you must be a member of the CG."

CG](https://www.w3.org/community/solid/join) if you’d like to contribute. People
agree to the [W3C Community License Agreement
(CLA)](http://www.w3.org/community/about/agreements/cla/) upon joining the CG.

The works in this repository use the [MIT
license](https://github.com/solid/specification/blob/main/LICENSE.md).

We accept different types of contributions, including some that don't require

> you to write a single line of code:

Reviewer: Missing list, or misplaced colon?

  or PR comments. You can apply suggested changes directly through the UI. You
  can make any other changes in your fork, then commit them to your branch.
* As you update your PR and apply changes, mark each conversation as
  [resolved](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/commenting-on-a-pull-request#resolving-conversations).
* If you run into any merge issues, checkout this [git
  tutorial](https://lab.github.com/githubtraining/managing-merge-conflicts) to
  help you resolve merge conflicts and other issues.
* If you are not the sole contributor, please identify all contributors in the
  PR comment. To add a contributor (other than yourself), mark them one per
  line as follows: `+@github_username`.

### Your PR is merged!

Congratulations :tada: Solid community thanks you :sparkles:.

## Publishing a technical report

This section describes how to prepare the technical reports and how they'll be
published on the Solid website.

### General guidance

* Content: the full content of the technical report must be human-readable with
  HTML alone. Any CSS and JavaScript should not interfere with the accessibility
  of the content.

> * Offline-friendly: there must not be any external dependencies (e.g., a network

Reviewer: ```suggestion
* Offline-friendly: there must not be any mandatory external dependencies (e.g., a network
```

Reviewer: (if that is what we mean)

# Solid Technical Reports Contributing Guide

> Thank you for investing your time into contributing to the Solid project!

Reviewer 2: ```suggestion
Thank you for investing your time in contribution to the Solid project!
```

search open issues to see if someone else has reported the same thing. If it's
something new, open an issue. We'll use the issue to have a conversation about
the problem you want to fix.

### Pull requests

A [pull
request](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/about-pull-requests)
is a way to suggest changes in our repository.

### Translations

We aim to have our documents internationalized and available in multiple
languages. The source content in this repository is written in English. We
accept contributions to localize the English content.

## Contributions at source repository

Each CG work item listed at [Solid Technical
Reports](https://solidproject.org/TR/) has its own repository and any issues
and PRs should be made at their source.

The persistent version of technical reports published in this repository (see
[publishing technical reports](#publishing-technical-reports) are intended to
be read-only snapshots of the CG work items.

[Solid
Panels](https://github.com/solid/specification/blob/main/README.md#solid-panels)
focus on certain work streams, with an aim to propose technical reports for
the Solid ecosystem.

> To build shared understanding, [user
> stories](https://github.com/solid/user-stories) can also be submitted.

Reviewer 3: ```suggestion
A low-barrier way to add concrete suggestions is to submit a [user
story](https://github.com/solid/user-stories). User stories are descriptions of desired features written in a special way, see the description there for details. User stories may then be used when formulating use cases and requirements.
```

Reviewer 3: I decided I needed to figure out how to make concrete suggestions, so there you go!
